petclinic.yaml

```diff
@@ -11,16 +11,21 @@ tools:
     type: dockerImage
     image: maven:3.5.4-jdk-8
     entryPoint: sleep infinity
-    persistentVolumes:
+    volumes:
       - name: maven-repo
-        containerPath: /root/.m2
-    services:
+        mountPath: /root/.m2
+      - name: projects
+        mountPath: /projects
+    endpoints:
       - name: spring-boot
-        port: 8080
-    mountProjectsSources: true
+        public: true
+        targetPort: 8080
+        attributes:
+          protocol: http
+          secure: false
   - name: mysql
-    type: composeFile
-    local: docker-compose.yml
+    type: kubernetesListFile
```

Comment:
Why not `KubernetesList`?

Reply:
At the beginning, I use KubernetesList, but then I noticed that `compose` tools is `composeFile`. So, I assume that it makes sense to have KubernetesListFile which would have the following structure 
```yaml 
  name: mysql
  type: kubernetesListFile
  local: kubernetes-mysql.yml  // maybe we can find better name for this field
```

and KubernetesList tool with the following structure
```yaml
  name: mysql
  type: kubernetesList
  content: // maybe we can find better name for this field
    items:
      - kind: Pod
        ...
      - kind: Service
        ...
```
or 
```yaml
  name: mysql
  type: kubernetesList
  items:
    - kind: Pod
      ...
    - kind: Service
      ...
```

Do you think it makes more sense to have one tool type `KubernetesList` and have options to specify embeded content or reference file(maybe with http link)?

Reply:
also why does it include List in the name of the attribute ?

Reply:
It defines content model. List is a kind of object in Kubernetes/OpenShift https://github.com/kubernetes/kubernetes/blob/master/hack/testdata/list.yaml#L2

I think it's a good idea to specify `kubernetes`/`openshift` as tool type and then Che Server might resolve content model itself, like whether it is `List`/`OpenShiftTemplate` or just one `Pod`.

Reply:
yes I'm aware of `kind: List`inside a kubernetes file definition.

But for me, type should be `type: kubernetes` then in the kubernetes file (or the content) it'll be kind `List` (or it could be something else as well)

It's up to the implementation to decide if it supports or not the content of the file included

Reply:
@benoitf 
> But for me, type should be type: kubernetes then in the kubernetes file (or the content) it'll be kind List (or it could be something else as well)
> 
>It's up to the implementation to decide if it supports or not the content of the file included

Agree with you. Will change `kubernetesList` to `kubernetes`

Reply:
Is `kubernetes-mysql.yml` a regular kubernetes resource manifest? I think it should be. If that is the case, we are assuming(forcing???) that manifest to only include the resources that we want to be materialized as part of the workspace. This will be limiting for most projects as we will most likely encounter a helm chart or openshift template on the project. I think we should also have `selectors` in here to select the resources that we want.
```yaml
- name: mysql
  type: kubernetes
  local: application.yml
  selector: 
     name: mysql
- name: reddit
  type: kubernetes
  local: application.yml
  selector: 
     name: reddit-cache  
```